include/structured/Dialect/Indexing/IR/Indexing.h

```diff
@@ -18,7 +18,16 @@
 #include "structured/Dialect/Indexing/IR/IndexingOpsDialect.h.inc"
 
 namespace mlir {
-namespace indexing {} // namespace indexing
```

Comment:
I *think* that you shouldn't need these forward declarations (even if you needed the custom directive, see below). If it doesn't compile without these, you might miss an include to a generated header file...